File: dev/src/samples/NetSamplesVB/Utilities/ObjectIdEnumerableExtensions.vb
Imports System.Collections.Generic

Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.Civil.DatabaseServices
Imports Autodesk.Civil.DatabaseServices.Styles

Namespace Autodesk.CivilizedDevelopment
  Module ObjectIdEnumerableExtensions
    <System.Runtime.CompilerServices.Extension()> _
    Public Function GetStyle(ids As IEnumerable(Of ObjectId),
                             name As String) As ObjectId
      For Each id As ObjectId In ids
        Dim style As StyleBase =
          TryCast(id.GetObject(OpenMode.ForRead), StyleBase)
        If style.Name = name Then
          Return id
        End If
      Next
      Return ObjectId.Null
    End Function
  End Module

End Namespace

File: dev/src/samples/NetSamples/Utilities/ObjectIdEnumerableExtensions.cs
using System;
using System.Collections.Generic;

using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.Civil.DatabaseServices;
using Autodesk.Civil.DatabaseServices.Styles;

namespace Autodesk.CivilizedDevelopment
{
    internal static class ObjectIdEnumerableExtensions
    {
        public static ObjectId GetStyle(this IEnumerable<ObjectId> ids, 
            string name)
        {
            foreach (ObjectId id in ids)
            {
                StyleBase style = id.GetObject(OpenMode.ForRead) 
                    as StyleBase;
                if (style.Name == name)
                {
                    return id;
                }
            }
            return ObjectId.Null;
        }
    }
}